pkg/utils/injection/injection.go

```diff
@@ -0,0 +1,61 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package injection
+
+import (
+	"context"
+
+	"github.com/awslabs/karpenter/pkg/utils/options"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/rest"
+)
+
+type resourceKey struct{}
+
+func WithResource(ctx context.Context, namespacedname types.NamespacedName) context.Context {
+	return context.WithValue(ctx, resourceKey{}, namespacedname)
+}
+
+func GetResource(ctx context.Context) types.NamespacedName {
+	retval := ctx.Value(resourceKey{})
+	if retval == nil {
```

Comment:
Why is this if check necessary compared to e.g. line 45

Reply:
Good question. In fact there should be a nil check at line 45 as well. Or it will fail if GetOptions is called without calling WithOptions first. May need to beef up nil checking for all the places that call GetOptions.